eclair-core/src/main/scala/fr/acinq/eclair/Features.scala

```diff
@@ -19,63 +19,123 @@ package fr.acinq.eclair
 import scodec.bits.{BitVector, ByteVector}
 
 /**
-  * Created by PM on 13/02/2017.
-  */
+ * Created by PM on 13/02/2017.
+ */
+
+sealed trait FeatureSupport
+
+// @formatter:off
+object FeatureSupport {
+  case object Mandatory extends FeatureSupport
+  case object Optional extends FeatureSupport
+}
+// @formatter:on
+
+sealed trait Feature {
```

Comment:
Not too sure about this:

```suggestion
sealed trait Bolt9Feature {
```

Reply:
Not sure either, it's a bit verbose...